Source/Core/AudioCommon/WaveFile.cpp

```diff
@@ -138,6 +157,16 @@ void WaveFileWriter::AddStereoSamplesBE(const short *sample_data, u32 count)
 		conv_buffer[2 * i + 1] = Common::swap16((u16)sample_data[2 * i]);
 	}
 
+	if (sample_rate != current_sample_rate)
+	{
+		Stop();
+		file_index++;
+		std::stringstream filename;
+		filename << File::GetUserPath(D_DUMPAUDIO_IDX) << basename << file_index << ".wav";
```

Comment:
I don't think it would be, if someone automates joining audio with a script or something, having preset names would be better.